### PR TITLE
Feat/stepper poc and navigation button

### PR DESCRIPTION
**Pull request by jwinding**
## Feature description
Adds the new stepper logic, and two new components: navigation button and navigation button group.

## Solution description
Adds the logic for using the connection matrix into the form. Groups the form navigation methods into an object, generated by the form hook, which is then passed to the various places where it is needed.

## What areas is affected by these changes?.
The form container, various places that use the navigation, and the two new components.

## Is there any existing behaviour change of other features due to this code change?
Yes; how we navigate in the form. 

## Covered unit tests cases / E2E test cases?
No.

## Are your code structured in a way so that reviewers can understand it?
Yes, hopefully. The matrix logic is perhaps not very clearly explained, but if one has grasped the idea that the matrix element M[step1][step2] tells us how step1 is connected to step2, then it should be understandable. 

## Was this feature tested in the following environments?
- [x] On a iOS device/simulator.

There's an issue here: this code is part of a large branch that I've been working on all week, so this is like the 'first part' of a set, and they are closely connected. I'm trying to break it up into logically separate PRs, but this is not 100% possible. This means that it might be hard to test these features in isolation.

### CODE REVIEW

File: source/components/molecules/FormField/FormField.js [hunk 1]
```diff
@@ -9,6 +9,8 @@ import SubstepButton from '../SubstepButton';
 import colors from '../../../styles/colors';
 import DateTimePickerForm from '../DateTimePicker';
 
+import ButtonField from '../ButtonField/ButtonField';
```
Comment (TeddyAndersson) [inline source/components/molecules/FormField/FormField.js:12]: This component is removed and thereby causes storybook to not start, so i can't test the changes. The component export is also existing in molecules/index.js

File: .eslintrc [hunk 2]
```diff
@@ -41,6 +41,7 @@
       }
     ],
     "no-console": 0,
+    "no-undef": 0,
```
Comment (TeddyAndersson) [inline .eslintrc:44]: I guess this is a problem, but before we change this eslint. I think we would need to bring this rule up with the others in the team (y)

**Review by TeddyAndersson — CHANGES_REQUESTED** [review]
Can't test this PR in storybook due to that the component ButtonField is used in other components but the component is none existing

**Review by dannil76 — COMMENTED** [review]
Missing somekind of step by step description on how to test and what to expect.
It is difficult to give relevant feedback to the uninitiated without being particularly familiar with the matter.

File: source/components/atoms/StepperPOC/Stepper.stories.js [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+        description: 'test field',
+        id: 'text1',
+        text: 'navigate down to step 5',
+        color: 'green',
+      },
+    ],
+  },
+];
+
+storiesOf('New Stepper POC', module).add('default', () => (
```
[the remark this answers is not in the document]
Comment (jwinding) [reply source/components/atoms/StepperPOC/Stepper.stories.js]: Fixed.

File: source/components/molecules/FooterAction/FooterAction.js [hunk 4]
```diff
@@ -138,13 +136,15 @@ FooterAction.propTypes = {
...
    */
   answers: PropTypes.object,
-  /** Behaviour for the start action */
-  onStart: PropTypes.func,
-  /** Behaviour for the close action */
-  onClose: PropTypes.func,
-  /** Behaviour for the next page action */
-  onNext: PropTypes.func,
-  /** Behaviour for sending updates to context and/or backend */
+  formNavigation: PropTypes.shape({
```
[the remark this answers is not in the document]
Comment (jwinding) [reply source/components/molecules/FooterAction/FooterAction.js:142]: Added small comments describing the actions.

File: source/components/molecules/NavigationButtonField/NavigationButtonField.tsx [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+import styled from 'styled-components/native';
+import { Button, Icon, Text } from '../../atoms';
+import colors from '../../../styles/colors';
+
+const ButtonFieldWrapper = styled.View`
+  flex: 1;
+  margin: 2px;
+`;
+
+export type ActionType =
```
[the remark this answers is not in the document]
Comment (jwinding) [reply source/components/molecules/NavigationButtonField/NavigationButtonField.tsx]: Fixed.

File: source/components/molecules/FormField/FormField.js (hunk 1 above)
Comment (jwinding) [reply source/components/molecules/FormField/FormField.js]: Fixed. Removed references to ButtonField and replaced with NavigationButtonFIeld, so now the story works.

File: .eslintrc (hunk 2 above)
Comment (jwinding) [reply .eslintrc:44]: This is an issue with typescript; it complains when using certain primitive types, mainly Record.

File: source/components/atoms/StepperPOC/Stepper.stories.js [hunk 6]
```diff
@@ -0,0 +1,142 @@
+import React from 'react';
+import { storiesOf } from '@storybook/react-native';
+import StoryWrapper from '../../molecules/StoryWrapper';
+import Form from '../../../containers/Form/Form';
+
+const connectivityMatrix = [
```
[the remark this answers is not in the document]
Comment (jwinding) [reply source/components/atoms/StepperPOC/Stepper.stories.js:11]: Added descriptive comment explaining the logic and syntax of the matrix.

File: source/components/atoms/StepperPOC/Stepper.stories.js [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+  ['none', 'next', 'none', 'down', 'none', 'none'],
+  ['back', 'none', 'next', 'none', 'down', 'none'],
+  ['none', 'back', 'none', 'none', 'none', 'none'],
+  ['up', 'none', 'none', 'none', 'none', 'next'],
+  ['none', 'up', 'none', 'none', 'none', 'none'],
+  ['up', 'none', 'none', 'back', 'none', 'none'],
+];
+
+const steps = [
```
[the remark this answers is not in the document]
Comment (jwinding) [reply source/components/atoms/StepperPOC/Stepper.stories.js:20]: hmm, I think it's okay to keep it in the story file, it's not too much and really this data is the most important thing to look at here.

**Comment by jwinding** [comment]
> Missing somekind of step by step description on how to test and what to expect.
> It is difficult to give relevant feedback to the uninitiated without being particularly familiar with the matter.

Yeah, sorry, reading my description I understand that this is lacking, and in addition I failed at making the story work in the storybook, so this was really not good. 

Now I made the story work, so the thing to test is the Form Stepper story in storybook, where you can just navigate around a simple form with some navigation buttons and the usual next/back.

**Review by dannil76 — APPROVED** [review]

**Pull request merged**